.claude/hooks/gdd-permission-hook.sh

```
@@ -784,12 +838,141 @@ for _entry in ${redirect_commands[@]+"${redirect_commands[@]}"}; do
+#
+# The pattern `"$root/components/"*` also matches `$cwd == "$root/components/"`
+# itself (empty tail after the prefix strip), which would produce an
+# empty component name — Tier 3 would then treat the workspace-level
+# `components/` directory as a "component", emit a blank-component
+# nudge, and look up `realms/<realm>/adapters/.yaml`. Guard against
+# that by rejecting an empty first segment so the rule only fires
+# under a real `components/<comp>/` path.
+_ar_resolve_component() {
+    local cwd="$1" root="$2"
+    if [[ "$cwd" == "$root/components/"* ]]; then
+        local rest="${cwd#$root/components/}"
```

Shellcheck (0.11.0)

[info] Expansions inside ${..} need to be quoted separately, otherwise they match as patterns.

(SC2295)